src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -635,17 +896,24 @@
 
             var typeParameters1 = member1.GetMemberTypeParameters();
             var typeParameters2 = member2.GetMemberTypeParameters();
-            return HaveSameConstraints(typeParameters1, typeMap1, typeParameters2, typeMap2);
+            return HaveSameConstraints(typeParameters1, typeMap1, typeParameters2, typeMap2, typeComparison);
         }
 
-        public static bool HaveSameConstraints(ImmutableArray<TypeParameterSymbol> typeParameters1, TypeMap? typeMap1, ImmutableArray<TypeParameterSymbol> typeParameters2, TypeMap? typeMap2)
+        public static bool HaveSameConstraints(ImmutableArray<TypeParameterSymbol> typeParameters1, TypeMap? typeMap1, ImmutableArray<TypeParameterSymbol> typeParameters2, TypeMap? typeMap2, TypeCompareKind typeComparison)
         {
             Debug.Assert(typeParameters1.Length == typeParameters2.Length);
 
             int arity = typeParameters1.Length;
             for (int i = 0; i < arity; i++)
             {
-                if (!HaveSameConstraints(typeParameters1[i], typeMap1, typeParameters2[i], typeMap2))
+                TypeParameterSymbol typeParameter1 = typeParameters1[i];
+                TypeParameterSymbol typeParameter2 = typeParameters2[i];
+                if (!HaveSameConstraints(typeParameter1, typeMap1, typeParameter2, typeMap2, typeComparison))
+                {
+                    return false;
+                }
+                else if ((typeComparison & TypeCompareKind.IgnoreNullableModifiersForReferenceTypes) == 0
+                    && !HaveSameNullabilityInConstraints(typeParameter1, typeMap1, typeParameter2, typeMap2, typeComparison))
```

Comment:
>HaveSameNullabilityInConstraints [](http://example.com/codeflow?start=24&length=32)

It feels strange that this aspect is not handled by ```HaveSameConstraints``` check in the ```if``` above. Especially that both APIs are pubic and can be called independently. It would be unexpected that this API compares type parameters differently for the same TypeCompareKind.

Reply:
I agree, but that's the design of those two existing APIs. Redesigning them seems out-of-scope for this work

Reply:
> Redesigning them seems out-of-scope for this work

I do not agree. There should be no difference in behavior between this method and the method below. This API is should be just a loop that calls the other API, it should not change anything about comparing individual type parameters